js_modules/dagster-ui/packages/ui-core/src/live-data-provider/LiveDataProvider.tsx

```diff
@@ -38,11 +39,12 @@ export function useLiveData<T>(
 
   const [isRefreshing, setIsRefreshing] = React.useState(false);
 
-  // use JSON.stringify to avoid unsubscribing/re-subscribing to the same keys
-  const keysJSON = useMemo(() => JSON.stringify(keys), [keys]);
+  // Hash the keys and use that as a dependency to avoid unsubscribing/re-subscribing to the same keys in case the reference changes but the keys are the same
+  const keysHash = useMemo(() => hashObject(_keys), [_keys]);
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const keys = useMemo(() => _keys, [keysHash]);
```

Comment:
If we end up doing this more I could see wrapping this into a little helper so that this trick doesn't have to be repeated.

This seems like a nice solution to me.